src/messaging/ReliableMessageContext.h

```diff
@@ -164,18 +161,33 @@ class ReliableMessageContext
         /// When set, we have had Close() or Abort() called on us already.
         kFlagClosed = (1u << 7),
 
-        /// When set, signifies that the exchange is requesting Sleepy End Device active mode.
-        kFlagActiveMode = (1u << 8),
-
         /// When set, signifies that the exchange created sorely for replying a StandaloneAck
-        kFlagEphemeralExchange = (1u << 9),
+        kFlagEphemeralExchange = (1u << 8),
 
         /// When set, ignore session being released, because we are releasing it ourselves.
-        kFlagIgnoreSessionRelease = (1u << 10),
+        kFlagIgnoreSessionRelease = (1u << 9),
+
+        // When set, sender knows that the peer (receiver) should currently be active
+        // When bit is not set, we don't know if the peer (receiver) is active or not
+        kPeerShouldBeActive = (1u << 10)
     };
 
     BitFlags<Flags> mFlags; // Internal state flags
 
+    /**
+     * Function checks with information available in the ExchangeContext if the peer should be active or not
+     * API should only be used when we are determining if we need to use the Active or Idle retransmission timeout
+     *
+     * If we are not initiator                          -> peer is active
+     * If ephemeral context                             -> peer is active
+     * else we default to information provided by the client
+     *
+     * @return true Based on available information, Peer should be active
+     * @return false We weren't able to determine if the peer is active.
+     *               We don't know in which state the peer is.
+     */
+    bool IsPeerLikelyActiveHint();
```

Comment:
Why is this API both here and on exchangecontext?

Reply:
shouldn't, removing it.